controllers/numaresourcesoperator_controller.go

```diff
@@ -305,6 +311,46 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
 	return daemonSetsNName, nil
 }
 
+func (r *NUMAResourcesOperatorReconciler) deleteUnusedDaemonSets(ctx context.Context, instance *nropv1alpha1.NUMAResourcesOperator, mcps []*machineconfigv1.MachineConfigPool) []error {
+	klog.V(3).Info("Delete DS start")
+	var errors []error
+	var daemonSetList appsv1.DaemonSetList
+	if err := r.List(ctx, &daemonSetList, &client.ListOptions{Namespace: instance.Namespace}); err != nil {
+		klog.Infof("error while getting DS list: %v", err)
+		return append(errors, err)
+	}
+
+	// Generate the names of the DaemonSets that should be running,
+	// one for each MachineConfigPool
+	expectedDaemonSetNames := sets.NewString()
+	for _, mcp := range mcps {
+		expectedDaemonSetNames = expectedDaemonSetNames.Insert(rtestate.GetComponentName(instance.Name, mcp.Name))
+	}
+
+	for _, ds := range daemonSetList.Items {
+		if !expectedDaemonSetNames.Has(ds.Name) {
+			if r.isOwnedBy(ds.GetObjectMeta(), instance) {
+
+				if err := r.Client.Delete(ctx, &ds); err != nil {
+					klog.Infof("error while deleting [%s]: %v", ds.Name, err)
+					errors = append(errors, err)
```

Comment:
I'm not sure we want to accumulate errors here, need to think about it.
If we are not sure this case is likely and we really want to keep iterating and delete unused nodegroups, I'd start very simple and just bail out on the first error.

Reply:
Well I just thought of accumulate them because an error while deleting a DS could stop the process preventing the deletion of the others that may work properly. Also, even if all of them fail, it could be interesting to have all the information at once, unless the deletion operation would take too much resources does not seem a bad exchange.
Also I start from the base that an error while deleting DS resources will be just a "warning" at operator level as this is just to improve resource utilization but no interferes with the operator functionality.

Reply:
I need to think more about this

Reply:
I'm still not sold, but let's keep the accumulation for now.